### PR TITLE
Add yaml libs reference to HTTP proxy test suite.

### PR DESCRIPTION
**Pull request by randall**
Co-authored-by: Jean Baptiste Favre <debian@jbfavre.org>
(cherry picked from commit c0c0c8051241a02fee5d06d0a6993d1be90834ee)

Conflicts:
    proxy/http/Makefile.am

### CODE REVIEW

**Comment by zwoop** [comment]
[approve ci]

**Comment by randall** [comment]
Pretty sure the autest failures are unrelated:

File: proxy/http/Makefile.am [hunk 1]
```diff
@@ -158,7 +160,8 @@ test_HttpTransact_LDADD = \
 	@LIBZ@ \
 	@LIBLZMA@ \
 	@OPENSSL_LIBS@ \
-	-lm
+	-lm \
+	@YAMLCPP_LIBS@
```
Comment (jpeach) [inline proxy/http/Makefile.am:164]:
```suggestion
 	@YAMLCPP_LIBS@ \
	-lm
```

BEst of keep `-lm` last since order matters :)

**Comment by bneradt** [comment]
We'll need to pull in a variety of test updates to 9.2.x to work with fedora:38 and OpenSSL 3.0. I can work on that this afternoon.

**Comment by bneradt** [comment]
> We'll need to pull in a variety of test updates to 9.2.x to work with fedora:38 and OpenSSL 3.0. I can work on that this afternoon.

This should help the 9.2.x branch run the autests:
https://github.com/apache/trafficserver/pull/9952

**Comment by bneradt** [comment]
[approve ci autest]

**Review by bneradt — APPROVED** [review]

File: proxy/http/Makefile.am (hunk 1 above)
Comment (randall) [reply proxy/http/Makefile.am:164]: I'm just a vessel for the PR. This is a backport of #9885 to 9.2.x. We should follow up with a another PR to make it last.

**Review by zwoop — APPROVED** [review]

**Pull request merged**